File: 프로그래머스/2/42747. H－Index/H－Index.js
/*
h-index : 발표한 논문 수, 인용횟수 둘 중 최소값
*/

function solution(citations) {
    citations.sort((a, b) => a - b);
    
    const n = citations.length;
    for (let i = 0; i < n; i ++) {
        const citation = citations[i];
        const restCite = n - i;
        
        if (citation >= restCite) {
            return restCite;
        }
    }
    
    return 0;
}

File: 프로그래머스/2/42747. H－Index/H－Index.py
import heapq

def solution(citations):
    n = len(citations)
    
    h = []
    
    for citation in citations:
        heapq.heappush(h, citation)
        
    while h:
        min_citation = heapq.heappop(h)
        
        if min_citation >= len(h) + 1:
            return len(h) + 1
        
    return 0